python/semantic_kernel/contents/utils/data_uri.py

```diff
@@ -0,0 +1,122 @@
+        else:
+            self.data_bytes = value
+
+    @model_validator(mode="before")
+    @classmethod
+    def _validate_data(cls, values: dict[str, Any]) -> dict[str, Any]:
+        """Validate the data."""
+        if not values.get("data_bytes") and not values.get("data_str"):
+            raise ContentInitializationError("Either data_bytes or data_str must be provided.")
+        return values
+
+    @model_validator(mode="after")
+    def _parse_data(self) -> Self:
+        """Parse the data bytes to str."""
+        if not self.data_str and self.data_bytes:
+            if self.data_format and self.data_format.lower() == "base64":
+                self.data_str = base64.b64encode(self.data_bytes).decode("utf-8")
+            else:
+                self.data_str = self.data_bytes.decode("utf-8")
+        if self.data_format and self.data_format.lower() == "base64" and self.data_str:
+            try:
+                if not self.data_bytes:
+                    self.data_bytes = base64.b64decode(self.data_str, validate=True)
+                else:
+                    base64.b64decode(self.data_str, validate=True)
+            except binascii.Error as exc:
+                raise ContentInitializationError("Invalid base64 data.") from exc
+        return self
+
+    @classmethod
+    def from_data_uri(cls: type[_T], data_uri: str | Url, default_mime_type: str = "text/plain") -> _T:
+        """Create a DataUri object from a data URI string or pydantic URL."""
+        if isinstance(data_uri, str):
+            try:
+                data_uri = Url(data_uri)
+            except ValidationError as exc:
+                raise ContentInitializationError("Invalid data uri format.") from exc
+
+        data = data_uri.path
+        if not data or "," not in data:
+            raise ContentInitializationError("Invalid data uri format. The data is missing.")
+
+        pattern = "(((?P<mime_type>[a-zA-Z]+/[a-zA-Z-]+)(?P<parameters>(;[a-zA-Z0-9]+=+[a-zA-Z0-9]+)*))?(;+(?P<data_format>.*)))?(,(?P<data_str>.*))"  # noqa: E501
+        match = re.match(pattern, data)
+        if not match:
+            raise ContentInitializationError("Invalid data uri format.")
+        matches = match.groupdict()
+        if not matches.get("data_format"):
+            matches.pop("data_format")
+        if not matches.get("parameters"):
+            matches.pop("parameters")
+        else:
+            matches["parameters"] = matches["parameters"].strip(";").split(";")
+        if not matches.get("mime_type"):
+            matches["mime_type"] = default_mime_type
+        return cls(**matches)
+
+    @field_validator("parameters", mode="before")
+    def _validate_parameters(cls, value: list[str] | dict[str, str] | None = None) -> dict[str, str]:
+        if not value:
+            return {}
+        if isinstance(value, dict):
+            return value
+
+        new: dict[str, str] = {}
+        for item in value:
+            if item.strip() == "":
```

Comment:
Would it make sense to do the follow up string manipulation on the stripped value? For example, could `item` be `"key1=val1   "`, then `item.split("=", maxsplit=1)` would return `["key1", "val1   "]`. Where if we store `item.strip()` in a variable and use it in line 103, 105, and 107, we get rid of that padding.